pkgs/development/interpreters/octave/build-octave-package.nix

```diff
@@ -123,6 +123,10 @@ stdenv.mkDerivation (
       else
         preBuild;
 
+    # Add Octave's shared objects to ld library path for some packages which
+    # need to link their C/C++/Fortran code against Octave's internals.
+    LD_LIBRARY_PATH = lib.makeLibraryPath nativeBuildInputs';
+
```

Comment:
I'm a little confused about how `LD_LIBRARY_PATH` helps.

however, was able to get the build to work with this
```diff
diff --git a/pkgs/development/octave-modules/ltfat/default.nix b/pkgs/development/octave-modules/ltfat/default.nix
index 43fe020b322a..d22a78938b5a 100644
--- a/pkgs/development/octave-modules/ltfat/default.nix
+++ b/pkgs/development/octave-modules/ltfat/default.nix
@@ -9,6 +9,7 @@
   lapack,
   blas,
   portaudio,
+  octave,
   jdk,
 }:
 
@@ -21,6 +22,8 @@ buildOctavePackage rec {
     sha256 = "sha256-FMDZ8XFhLG7KDoUjtXvafekg6tSltwBaO0+//jMzJj4=";
   };
 
+  NIX_LDFLAGS = "-L${lib.getLib octave}/lib/octave/9.3.0";
+
   buildInputs = [
     fftw
     fftwSinglePrec
```

on darwin and linux tho didn't test the plugin to see if it actually works -- there were some linker warnings on darwin.

Reply:
ok looking at the bundle in `oct/Makefile_unix` it seems to export `LDFLAGS`

```
export LDFLAGS := -L$(shell $(MKOCTFILE) -p LIBDIR) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS)
```
and thru some experimentation when `LDFLAGS` are set I guess that overrides the flags used by `mkoctfile` because
```
make -f Makefile_unix comp_atrousfilterbank_td.oct
mkoctfile -strip -I../src/modules/libltfat/include -DNDEBUG -L../lib -lltfat comp_atrousfilterbank_td.cc
mkoctfile: stripping disabled on this platform
ld: library not found for -loctinterp
```
but just can comment out `LDFLAGS` in the makefile and then the command succeeds.

so this seems like an upstream issue.

Reply:
ok -- apologies for all the posts. this seems like a workaround until the problem is fixed in the upstream makefile.

```diff
diff --git a/pkgs/development/octave-modules/ltfat/default.nix b/pkgs/development/octave-modules/ltfat/default.nix
index 43fe020b322a..d98253467190 100644
--- a/pkgs/development/octave-modules/ltfat/default.nix
+++ b/pkgs/development/octave-modules/ltfat/default.nix
@@ -1,4 +1,5 @@
 {
+  stdenv,
   buildOctavePackage,
   lib,
   fetchurl,
@@ -32,6 +33,12 @@ buildOctavePackage rec {
     jdk
   ];
 
+  # work around issue in upstream oct/Makefile_unix where it doesn't add
+  # -L$(mkoctfile -p OCTLIBDIR) to exported LDFLAGS
+  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    NIX_LDFLAGS+=" -L$(mkoctfile -p OCTLIBDIR)"
+  '';
+
   meta = with lib; {
     name = "The Large Time-Frequency Analysis Toolbox";
     homepage = "https://octave.sourceforge.io/ltfat/index.html";

```

Reply:
ok -- sorry for all the posts. I just realized that can use `patches` in the octave builder. `LDFLAGS` in the env will clobber `mkoctfile` ldflags, so make sure to call `mkoctfile -p LDFLAGS` first.

should file an issue: https://github.com/ltfat/ltfat/issues

<details><summary>don't clobber ldflags.diff</summary>

```diff
diff --git a/pkgs/development/octave-modules/ltfat/append-ldflags.diff b/pkgs/development/octave-modules/ltfat/append-ldflags.diff
new file mode 100644
index 000000000000..f0f15bc1b5fe
--- /dev/null
+++ b/pkgs/development/octave-modules/ltfat/append-ldflags.diff
@@ -0,0 +1,13 @@
+diff --git a/oct/Makefile_unix b/oct/Makefile_unix
+index 21e026b..c9d767a 100644
+--- a/oct/Makefile_unix
++++ b/oct/Makefile_unix
+@@ -36,7 +36,7 @@ endif
+ export CXXFLAGS := $(shell $(MKOCTFILE) -p CXXFLAGS) -std=gnu++11 -Wall -DLTFAT_LARGEARRAYS $(OPTCXXFLAGS)
+ # export is necessary, otherwise LFLAGS won't have any effect
+ # at least on Windows and on Mac
+-export LDFLAGS := -L$(shell $(MKOCTFILE) -p LIBDIR) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS) 
++export LDFLAGS := $(shell $(MKOCTFILE) -p LDFLAGS) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS)
+ undefine LFLAGS
+ unexport LFLAGS
+ 
diff --git a/pkgs/development/octave-modules/ltfat/default.nix b/pkgs/development/octave-modules/ltfat/default.nix
index 43fe020b322a..c5462f6392ea 100644
--- a/pkgs/development/octave-modules/ltfat/default.nix
+++ b/pkgs/development/octave-modules/ltfat/default.nix
@@ -32,6 +32,11 @@ buildOctavePackage rec {
     jdk
   ];
 
+  patches = [
+    # don't clobber mkoctfile ldflags, append them
+    ./append-ldflags.diff
+  ];
+
   meta = with lib; {
     name = "The Large Time-Frequency Analysis Toolbox";
     homepage = "https://octave.sourceforge.io/ltfat/index.html";

```
</details>